keanu-python/tests/test_vertex.py

```diff
@@ -240,23 +249,35 @@ def test_you_can_set_and_cascade(ctor, args, expected_type, value, assert_vertex
     assert_vertex_value_equals(vertex2, expected_type, value)
 
     two_values_are_equal = equal_vertex.get_value()
-    assert two_values_are_equal.dtype == np.bool_
-    assert two_values_are_equal.shape == vertex1.get_value().shape == vertex2.get_value().shape
-    assert np.all(two_values_are_equal)
+    is_scalar = type(two_values_are_equal) == bool
+    assert type(two_values_are_equal) == bool or two_values_are_equal.dtype == np.bool_
+
+    if not is_scalar:
+        assert two_values_are_equal.shape == vertex1.get_value().shape == vertex2.get_value().shape
+        assert np.all(two_values_are_equal)
 
     two_values_are_not_equal = not_equal_vertex.get_value()
-    assert two_values_are_not_equal.dtype == np.bool_
-    assert two_values_are_not_equal.shape == vertex1.get_value().shape == vertex2.get_value().shape
-    assert np.all(np.invert(two_values_are_not_equal))
+    assert type(two_values_are_equal) == bool or two_values_are_not_equal.dtype == np.bool_
+
+    if not is_scalar:
+        assert two_values_are_not_equal.shape == vertex1.get_value().shape == vertex2.get_value().shape
+        assert np.all(np.invert(two_values_are_not_equal))
+
+
+@pytest.mark.parametrize("ctor, args, expected_type", [(Gaussian, (0., 1.), float), (UniformInt, (0, 10), int),
+                                                       (Bernoulli, (0.5,), bool)])
+@pytest.mark.parametrize("value, assert_vertex_value_equals", [(4, assert_vertex_value_equals_scalar),
+                                                               (5., assert_vertex_value_equals_scalar),
+                                                               (True, assert_vertex_value_equals_scalar)])
+def test_you_can_set_and_cascade_scalar(ctor, args, expected_type, value, assert_vertex_value_equals):
+    test_you_can_set_and_cascade(ctor, args, expected_type, value, assert_vertex_value_equals)
```

Comment:
This is a bit hard to read. I think it'll be worth rewriting the test for scalars.

Reply:
Also you can call `assert_vertex_value_equals_scalar` directly here instead of passing it as a parameter

Reply:
👍